sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileTable.scala

```diff
@@ -46,17 +46,27 @@ abstract class FileTable(
       sparkSession, rootPathsSpecified, caseSensitiveMap, userSpecifiedSchema, fileStatusCache)
   }
 
-  lazy val dataSchema: StructType = userSpecifiedSchema.orElse {
-    inferSchema(fileIndex.allFiles())
-  }.getOrElse {
-    throw new AnalysisException(
-      s"Unable to infer schema for $name. It must be specified manually.")
-  }.asNullable
+  lazy val dataSchema: StructType = userSpecifiedSchema.map { schema =>
+      val partitionSchema = fileIndex.partitionSchema
+      val equality = sparkSession.sessionState.conf.resolver
```

Comment:
`equality` -> `resolver`?

Reply:
The naming is following DataSource.scala line 185. I think it is OK.

Reply:
There is only [one line](https://github.com/apache/spark/blame/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala#L111) written two year ago. All the other new instances use `resolver = sparkSession.sessionState.conf.resolver` (more than 7).

Reply:
If you search with `conf.resolver`, there are more instances.